### ISSUE TITLE
Are the loss functions behaving correctly?

### ISSUE THREAD

**Issue by davibarreira** [FluxML/model-zoo#286]
I've written the following simple code for training a NN:
```julia
function NN()
    return Chain(
            Dense(2, 10,relu),
            Dense(10,1,x->σ.(x))
            )
end
train_size = 1000
m    = NN()
ps = Flux.params(m)
opt = Descent(0.01)

real = generate_real_data(train_size)
fake = generate_fake_data(train_size)
X    = hcat(real,fake)
Y    = vcat(ones(train_size),zeros(train_size))
data = Flux.Data.DataLoader(X, Y, batchsize=100);
loss(x, y) = sum(Flux.Losses.crossentropy(m(x), y))
Flux.train!(loss, ps, data, opt)
```

Now, if I use any of the loss functions (i.e. `logitcrossentropy`, `binarycrossentropy`, `logitbinarycrossentropy`) the model does not properly train. Is this behaviour correct? And if so, why is that? Because it would seem to me that they all should work.
